mathesar_ui/src/component-library/toast/ToastItem.svelte

```diff
@@ -0,0 +1,70 @@
+<script lang="ts">
+  import { faTimes } from '@fortawesome/free-solid-svg-icons';
+  import { Icon } from '..';
+  import { ensureReadable } from '../common/utils/storeUtils';
+  import type { ToastEntry } from './ToastController';
+  
+  export let entry: ToastEntry;
+
+  $: ({ props, controller } = entry);
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  $: ({ progress, dismiss } = controller);
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  $: ({ pause, resume } = progress);
+
+  $: readableTitle = ensureReadable(props.title);
```

Comment:
It would be better to replace `ensureReadable` with something like `getFromReadable` (we need a better name).

`ensureReadable` creates a store if it's not a store already. I like having this util method but I don't see value in using it here. It only adds time & space complexity of creating of multiple stores for a usecase which only requires the value of the stores.

Instead, `getFromReadable` would return the value of the store if a store is passed, or would return the value directly if the passed argument is not a store.

Reply:
I'm not really seeing how your `getFromReadable` function could work. I made the changes that I _think_ you're describing, and I put them in the branch `758_toast__getFromReadable`. Now when I go to Storybook > Systems > Toast > "High Level" > "Interactive" > "Dynamic content", I'm no longer able to update the toast content. That makes sense because the `props.message` store _object_ is not changing so the reactive statement `$: readableMessage = getFromReadable(props.message);` will not get triggered. The store _value_ is changing, and we can react to those changes via a subscription. But we can't subscribe to it unless we know it's already a store. If you're seeing a way to accomplish this, then I'd love to learn.

Further, I don't think the toast system warrants the level of performance optimization that you're seeking here. From what you've already taught me about Svelte stores, I understand that creating a new store incurs some performance costs. Here we are creating 7 additional stores per toast message. How much would you predict that those 7 additional stores would increase the total time to display one toast message? My guess is that it's not more than 2x -- meaning that in the worst case scenario, the additional _could_ double the time it would take to display one toast message, but not any more. And I can't imagine a scenario in which we'd ever be creating more than a few toast messages per second. So in a pragmatic sense, I don't see a performance issue here. Am I missing something?

It's worth noting that we _could_ eliminate the "Dynamic content" feature altogether. We'd still be able to update the content of a toast message, we'd just need to throw up a whole new toast message each time we make an update (using the same `id` of course). That's more cumbersome but still totally possible, and "dynamic content" is probably a rare use-case anyway. I'm fine with ripping this part out of the toast system. We wouldn't have the extra stores, and it would simplify things a bit.

All that said, I still really am curious if there's actually a way to implement the `getFromReadable` function you mentioned. Even if we eliminate the "Dynamic content" feature, I'd still like to know, just for the purpose of learning.

Reply:
I didn't think it through for Dynamic content. It would be essential to go with `ensureReadable` for it to work, since the compiler can only ensure store reactivity within the component based on appended the `$` symbol.